### DISCUSSION TITLE
How to add custom ClassTransformer to TeaVM gradle plugin?

### DISCUSSION THREAD

**Discussion by McMurat** [konsoletyper/teavm#985] [Q&A]
Hi! I want to write my own ClassTransformer to mutate the constructor of the java.lang.Object class using TeaVM. Currently I am using the Gradle plugin to generate a JS bundle. How can I add a custom ClassTransformer to the gradle TeaVM tool generating the JS code?
Thank you.

**Comment by konsoletyper** [comment]
You can write a custom TeaVM plugin. Just implement `TeaVMPlugin` interface and use `ServiceLoader` convention to register it. No documentation, no explanation, just clone respository, open in IDEA and find all implementations of `TeaVMPlugin` for reference. Also, you can try to search through issues/discussions for this topic
